mount/mount.go

```diff
@@ -22,3 +28,48 @@ func All(mounts []Mount, target string) error {
 	}
 	return nil
 }
+
+// MountsOption are option for mount. It indicate that we will mount Mounts to TmpParentDir/TmpDirPrefix
+// for TmpParentDir and TmpDirPrefix, please see ioutil.TempDir.
+type MountsOption struct {
```

Comment:
I think this is a bad name and makes this change confusing.  These are not really options but treated as an atomic mount.  Maybe find a better name for this?

Reply:
Maybe something like `TempMount` as the `Tmp*` strings are not really optional and this is just being used for temp mounts.

Reply:
>I think this is a bad name and makes this change confusing. These are not really options but treated as an atomic mount. Maybe find a better name for this?

What do you think "MountSpec" or "MountRequest"?

>Maybe something like TempMount as the Tmp* strings are not really optional and this is just being used for temp mounts.

I see . There are not cases that we specify the MountOption.TmpParentDir in containerd until now. 
You think that we only should let user to specify "mounts" and  "name" like "ctd-username","extract-" etc.
Or only specify "mounts".
I prefer to have "name", it is handy for debug.
What do  you think?

Reply:
What about `TransientMount` ? I think it's fine for `ParentDir` to be there even if not set, it will use the default, same as the `TempDir` function.

Reply:
The TempDir is unnecessary. Because user don't use it. So user should not care it.

Reply:
Administrators may want to put them somewhere else though, separated from the rest of the system for easy cleanup.

One cannot just `rm -rf /tmp/*` nowadays ;-)

Reply:
make sense.  @crosbymichael  WDYT?

Reply:
TransientMount or TempMount works for me.

Reply:
We use TempMount. It will be consistent with the prefix of field TmpParentDir.

Reply:
done